lib/mayaUsd/ufe/UsdUndoMaterialCommands.cpp

```diff
@@ -302,63 +363,17 @@ void UsdUndoAssignNewMaterialCommand::execute()
-        auto stageHierarchy = Ufe::Hierarchy::hierarchy(Ufe::Hierarchy::createItem(stagePath));
-        if (!stageHierarchy) {
+        auto stageItem
+            = UsdSceneItem::create(MayaUsd::ufe::stagePath(stage), stage->GetPseudoRoot());
+        auto createMaterialsScopeCmd = UsdUndoCreateMaterialsScopeCommand::create(stageItem);
+        if (!createMaterialsScopeCmd) {
             markAsFailed();
             return;
         }
+        createMaterialsScopeCmd->execute();
+        _cmds->append(createMaterialsScopeCmd);
 
-        // Find an available materials scope name.
-        // Usually the materials scope will simply have the default name (e.g. "mtl"). However, if
-        // that name is used by a non-scope object, a number should be appended (e.g. "mtl1"). If
-        // this name is not available either, increment the number until an available name is found.
-        Ufe::SceneItem::Ptr materialsScope = nullptr;
-        std::string         materialsScopeName = materialsScopeNamePrefix;
-        Ufe::SceneItemList  children = stageHierarchy->children();
-        for (size_t i = 1;; ++i) {
-            auto hasName = [&materialsScopeName](const Ufe::SceneItem::Ptr& item) {
-                return item->nodeName() == materialsScopeName;
-            };
-            auto childrenIterator = std::find_if(children.begin(), children.end(), hasName);
-            if (childrenIterator == children.end()) {
-                break;
-            }
-            if ((*childrenIterator)->nodeType() == "Scope") {
-                materialsScope = *childrenIterator;
-                break;
-            }
-
-            // Name is already used by something that is not a scope. Try the next name.
-            materialsScopeName = materialsScopeNamePrefix + std::to_string(i);
-        }
-
-        if (!materialsScope) {
-            auto createScopeCmd = UsdUndoAddNewPrimCommand::create(
-                UsdSceneItem::create(MayaUsd::ufe::stagePath(stage), stage->GetPseudoRoot()),
-                materialsScopeName,
-                "Scope");
-            if (!createScopeCmd) {
-                markAsFailed();
-                return;
-            }
-            createScopeCmd->execute();
-            _cmds->append(createScopeCmd);
-            auto scopePath = createScopeCmd->newUfePath();
-            // The code automatically appends a "1". We need to rename:
-            auto itemOps = Ufe::SceneItemOps::sceneItemOps(Ufe::Hierarchy::createItem(scopePath));
-            if (!itemOps) {
-                markAsFailed();
-                return;
-            }
-            auto rename = itemOps->renameItemCmd(Ufe::PathComponent(materialsScopeName));
-            if (!rename.undoableCommand) {
-                markAsFailed();
-                return;
-            }
-            _cmds->append(rename.undoableCommand);
-            materialsScope = rename.item;
-        }
+        auto materialsScope = createMaterialsScopeCmd->sceneItem(); 
```

Comment:
Factored out the creation of the materials scope into a new command, so that it can be reused by other commands in future work.